synapseclient/client.py

```diff
@@ -2263,6 +2262,17 @@ def submit(self, evaluation, entity, name=None, team=None, silent=False, submitt
         elif team:
             submission['submitterAlias'] = team.name
 
+        if isinstance(entity, synapseclient.DockerRepository):
+            submission['dockerRepositoryName'] = entity.repositoryName
+            docker_commits = self.restGET('/entity/{}/dockerCommit'.format(entity.id)) 
+            docker_digest = None
+            for commit in docker_commits['results']:
+                if dockerTag  == commit['tag']:
+                    docker_digest = commit['digest']
+                    submission['dockerDigest'] = docker_digest
+            if docker_digest is None:
```

Comment:
This check should be done before making the REST call.

Reply:
But the rest call list the commits (tag/digest pairs) for the given Docker repository.  I wouldn't be able to check this prior to the rest call.

Reply:
Oh I misread. Please check for dockerTag (the input parameter) is None before making the rest call.

Reply:
I got confused because of the error message. For dockerTag is None, the error message should explain what is wrong and what is the suggested behavior. For example: "A dockerTag is required to submit a DockerEntity."
For digest not found, the error message should first explain what went wrong, then what is the suggested behavior: "Disgest not found for tag: <xxx>. Please specify a docker tag that exists. The dockerTag 'latest' is used as default."

Reply:
Furthermore, the default value in the method signature does not guarantee that the value is not None in the function body. One can specify `submit( ..., dockerTag=None)` in their code. This seems to be an edge case, but our code should handle it anyway.

Reply:
Added these.